lib/matplotlib/backend_bases.py

```diff
@@ -2574,6 +2689,120 @@ class NonGuiException(Exception):
     pass
 
 
+class WindowEvent(object):
+    def __init__(self, name, window):
+        self.name = name
+        self.window = window
+
+
+class WindowBase(cbook.EventEmitter):
+    """The base class to show a window on screen.
+
+    Parameters
+    ----------
+    title : str
+        The title of the window.
+    """
+
+    def __init__(self, title, **kwargs):
+        super(WindowBase, self).__init__(**kwargs)
+
+    def show(self):
+        """
+        For GUI backends, show the figure window and redraw.
+        For non-GUI backends, raise an exception to be caught
+        by :meth:`~matplotlib.figure.Figure.show`, for an
+        optional warning.
+        """
+        raise NonGuiException()
+
+    def destroy(self):
+        """Destroys the window"""
+        pass
+
+    def set_fullscreen(self, fullscreen):
+        """Whether to show the window fullscreen or not, GUI only.
+
+        Parameters
+        ----------
+        fullscreen : bool
+            True for yes, False for no.
+        """
+        pass
+
+    def set_default_size(self, width, height):
+        """Sets the default size of the window, defaults to a simple resize.
+
+        Parameters
+        ----------
+        width : int
+            The default width (in pixels) of the window.
+        height : int
+            The default height (in pixels) of the window.
+        """
+        self.resize(width, height)
+
+    def resize(self, width, height):
+        """"For gui backends, resizes the window.
+
+        Parameters
+        ----------
+        width : int
+            The new width (in pixels) for the window.
+        height : int
+            The new height (in pixels) for the window.
+        """
+        pass
+
+    def get_window_title(self):
+        """
+        Get the title text of the window containing the figure.
+        Return None for non-GUI backends (e.g., a PS backend).
+
+        Returns
+        -------
+        str : The window's title.
+        """
+        return 'image'
```

Comment:
I don't really get this either, I just copied this from ``FigureManagerBase``... when I created this, I did it to conserve as much of the original functioning as possible.

I presume we did this so that non-interactive backends that didn't use their own ``FigureManagerX`` would report this to indicated that it just contained an image.

My personal preference here lies either in getting rid of this altogether, or creating a specific non-interactive ``WindowX`` class for use by those backends.